lib/statsd.js

```diff
@@ -38,20 +51,40 @@ const Client = function (host, port, prefix, suffix, globalize, cacheDns, mock,
     let socket;
     let errMessage;
 
-    if (args.protocol === 'tcp') {
+    if (args.protocol === TCP) {
       try {
         socket = net.connect(args.port, args.host);
         socket.setKeepAlive(true);
-      } catch (e) {
-        errMessage = `Could not establish connection to ${args.host}:${args.port}`;
-        if (instance.errorHandler) {
-          instance.errorHandler(new Error(errMessage));
-        } else {
-          console.log(errMessage);
+      } catch (err) {
+        errMessage = `Could not establish connection to ${args.host}:${args.port}: ${err}`;
+      }
+    } else if (args.protocol === UDS) {
+      if (! unixDgram) {
+        errMessage = 'The library unix_dgram, needed for UDS, is not available on your OS or version of Node.';
+      }
+      else {
+        const udsPath = args.path ? args.path : '/var/run/datadog/dsd.socket';
+        try {
+          socket = unixDgram.createSocket('unix_dgram');
+          socket.connect(udsPath);
+        } catch (err) {
+          errMessage = `Could not establish UDS connection to ${udsPath}: ${err}`;
         }
       }
     } else {
-      socket = dgram.createSocket('udp4');
+      try {
+        socket = dgram.createSocket('udp4');
+      } catch (err) {
+        errMessage = `Could not create socket: ${err}`;
+      }
+    }
+
+    if (errMessage) {
+      if (instance.errorHandler) {
+        instance.errorHandler(new Error(errMessage));
+      } else {
+        console.log(errMessage);
```

Comment:
nit: `console.error`?

Reply:
Oh I am terrible at just using console.log, sure I've done that elsewhere!  Will fix